### ISSUE TITLE
Implement Evictions

### ISSUE THREAD

**Issue by JoshOrndorff** [Off-Narrative-Labs/Tuxedo#98]
Background: #52 

UPDATE:
I have pivoted from what I described in the original issue description, toward an earlier and simpler design as described in the [first comment](https://github.com/Off-Narrative-Labs/Tuxedo/issues/98#issuecomment-1926570968).

Original Description:

Currently the tuxedo executive checks the verifiers on all inputs of all transactions. We previously considered the notion of "eviction" which was a way for a transaction to consume a UTXO without its verifier being satisfied.

As I described in the [comment on the background issue](https://github.com/Off-Narrative-Labs/Tuxedo/pull/52#issuecomment-1736135134), I think a better approach is to make the notion of Verifier less fundamental in the tuxedo executive. Rather than always checking verifiers in the core, instead we never check verifiers there. Instead we generalize the notion of ConstraintChecker like we did in #41 so that there are three levels like follows.

```Rust
/// The simplest and most common way to express transactions logic.
/// Verifiers are checked for you automatically.
/// Only the input data is passed along to the implementer.
/// The same one we've had from the beginning
trait SimpleConstraintChecker { /* --snip-- */ }

/// The more expressive but still reasonably safe way to express transaction logic.
/// Verifiers are checked for you automatically.
/// The entire UTXO including the already-checked Verifier are passed along to the implementer.
/// The one that was introduced in #41
trait ConstraintChecker { /* --snip-- */ }

/// The most expressive and dangerous way to express transaction logic.
/// Nothing is checked for you.
/// Implementers get the raw input data and do as they please.
/// Checking Verifiers is entirely optional.
/// This one is new and should be used sparingly.
/// If you want to write an eviction, this is the way to do it.
trait FullPowerConstraintChecker { /*--snip--*/ }

// Two blanket implementations instead of the one we currently have.

```

**Comment by JoshOrndorff** [comment]
Updates thoughts. Still similar to above. Add an enum

```rust
/// An input can be consumed in two way. It can be redeemed normally (probably with some signature, or proof) or it 
/// can be evicted. This enum is isomorphic to `Option<Vec<u8>>` but has more meaningful names
pub enum ConsumptionStrategy {
  /// The input is being consumed in the normal way with a signature or other proof provided by the spender.
  Redemption(Vec<u8>),
  /// The input is being forcefully evicted without satisfying its Verifier.
  Eviction,
```

Then change the input type:
```rust
pub struct Input {
    /// a reference to the output being consumed
    pub output_ref: OutputRef,
    // The method by which it is being consumed (eg by a signature)
    pub comsumption: ConsumptionStrategy,
}
```

Then in the case of the fully powerful constraint checker, the redeemed verifiers are still checked by core, and only information about which ones were evicted is passed along.

And finally, I'm not sure the middle level is actually necessary. The way sudo should work is by peeking at the utxo that holds the sudo key and providing a signature made by that key to the constraint checker. They way royalties should work is that they need to be claimed. Again the signature checking can happen at the constraint checker level.

**Comment by JoshOrndorff** [comment]
Solved in #196
